src/lib/interactionManager/flow.ts

```diff
@@ -0,0 +1,27 @@
+import { Interaction } from './interaction'
+import {
+  JWTEncodable,
+} from 'jolocom-lib/js/interactionTokens/JSONWebToken'
+import { InteractionType } from 'jolocom-lib/js/interactionTokens/types'
+
+export abstract class Flow {
+  protected ctx: Interaction
+  protected tokens: Array<JWTEncodable> = []
+
+  constructor(ctx: Interaction) {
+    this.ctx = ctx
+  }
+
+  // Can this abstract anything?
+  abstract async handleInteractionToken(
+    token: JWTEncodable,
+    messageType: InteractionType
+  ): Promise<any>
+
+  // @TODO Make sure this is only used for rendering
```

Comment:
It is, the return type must be specified though.

Reply:
`| | | ` vs ` < .. > `?

Reply:
Based on the small chat we had earlier, `<..>` (perhaps in the form of making the calss generic over it's state) would make adding new flows much easier.

Definetily not `| | |` though :D